src/ext/java/org/opentripplanner/ext/emissions/EmissionsFilter.java

```diff
@@ -45,7 +45,17 @@ public List<Itinerary> filter(List<Itinerary> itineraries) {
 
       calculateCo2EmissionsForCar(carLegs)
         .ifPresent(co2 -> {
-          itinerary.setEmissionsPerPerson(new Emissions(co2));
+          if (transitLegs.isEmpty()) {
+            itinerary.setEmissionsPerPerson(new Emissions(co2));
+          } else if (
+            itinerary.getEmissionsPerPerson() != null &&
+            itinerary.getEmissionsPerPerson().getCo2() != null
+          ) {
+            // Calculate park and ride emissions.
+            itinerary.setEmissionsPerPerson(
+              new Emissions(itinerary.getEmissionsPerPerson().getCo2().plus(co2))
+            );
```

Comment:
I wonder if this is better or keeping track of the total emissions within this class and then at the end adding the emissions to the itinerary.

Reply:
I agree. Please use a separate accumulator rather than repeatedly mutating the itinerary.